ci/Jenkinsfile

```diff
@@ -0,0 +1,148 @@
+    agent any
+    options {
+        disableConcurrentBuilds(abortPrevious: true)
+    }
+    stages {
+        stage('Checks & Builds') {
+            failFast true
+            parallel {
+                stage('Check') {
+                    options {
+                        timeout(time: 1, unit: 'HOURS')
+                    }
+                    environment {
+                        HOME = "${WORKSPACE}"
+                    }
+                    agent {
+                        docker {
+                            image 'gpuci/rapidsai-driver:21.10-cuda11.4-devel-ubuntu20.04-py3.8'
+                            label 'cpu'
+                        }
+                    }
+                    steps {
+                        cleanWs(
+                            deleteDirs: true,
+                            externalDelete: 'sudo rm -rf %s'
+                        )
+                        checkout scm
+                        sh "./ci/scripts/jenkins_checks.sh"
+                    }
+                }
+                stage('Build:linux:x86_64:gcc') {
+                    options {
+                        timeout(time: 4, unit: 'HOURS')
+                    }
+                    environment {
+                        PARALLEL_LEVEL = '16'
+                        PYTHON_VER = '3.8'
+                        RAPIDS_VER = '21.10'
+                        CUDA_VER = '11.4'
+                        USE_SCCACHE = '1'
+                        HOME = "${WORKSPACE}"
+                    }
+                    agent {
+                        docker {
+                            image 'gpuci/rapidsai-driver:21.10-cuda11.4-devel-ubuntu20.04-py3.8'
+                            label 'cpu4'
+                        }
+                    }
+                    steps {
+                        cleanWs(
+                            deleteDirs: true,
+                            externalDelete: 'sudo rm -rf %s'
+                        )
+                        checkout scm
+                        withCredentials([[
+                            $class: 'AmazonWebServicesCredentialsBinding',
+                            credentialsId: "aws-s3-gpuci",
+                            accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+                            secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+                        ]])
+                        {
+                            sh "./ci/scripts/jenkins_build.sh"
+                        }
+                    }
+                }
+            }
+        }
+        stage('Post Build Stages') {
+            failFast true
+            parallel {
+                stage('Test') {
+                    options {
+                        timeout(time: 1, unit: 'HOURS')
+                    }
+                    environment {
+                        PYTHON_VER = '3.8'
+                        RAPIDS_VER = '21.10'
+                        CUDA_VER = '11.4'
+                        HOME = "${WORKSPACE}"
```

Comment:
I see some of these variables are also used in the previous stages. You could move them to a top-level `environment` block if you want to DRY up the Jenkinsfile a bit.